### PR TITLE
fix(health-check): honor disabled voice config

### LINKED ISSUE #2349: health-check reports disabled voice-agent as a hard failure
## Problem

`startup-runtime.sh` intentionally sets `SKIP_VOICE=1` when neither `GEMINI_VOICE_API_KEY` nor `GEMINI_API_KEY` is configured, but `health-check.py::run_all_checks()` always treats port 9900 as a required service.

That makes a supported web-only / credential-free setup permanently unhealthy.

## Reproduction

On a checkout whose resolved `.env` contains no Gemini voice key:

```text
$ bash -c 'source src/startup-runtime.sh; configure_startup_runtime; printf "SKIP_VOICE=%s\n" "$SKIP_VOICE"'
  ~ voice agent disabled (set GEMINI_VOICE_API_KEY or GEMINI_API_KEY to enable)
SKIP_VOICE=1
```

The same host reports:

```text
$ python3 src/health-check.py
✗ voice-agent  down  port 9900
...
1 issue(s) found:
  - voice-agent: down (port 9900)
```

`health-check.py --fix` cannot repair it because there is intentionally no configured voice service and no `com.sutando.voice-agent` plist.

## Expected

Health-check should apply the same voice-configured decision as startup:

- no voice key / explicit `SKIP_VOICE=1` → voice-agent, voice-watchers, and voice-transport are optional/disabled, not failures;
- a configured voice key → retain the current required port and transport checks;
- malformed/unreadable config should fail visibly rather than silently claim voice is disabled.

The decision should use the canonical resolved `.env` path and preserve environment-over-file precedence.

## Scope / risk

This is reporting only; no service behavior needs to change. The main risk of a fix is hiding a genuinely broken configured voice agent, so regression coverage should include configured, unconfigured, explicit skip, and environment-override cases.

### PR DESCRIPTION
**Pull request by john-the-dev**
## What changed, and why?

Health-check now resolves whether the voice stack is configured before treating port 9900, its watcher registration, and its Gemini transport as required. Missing credentials or `SKIP_VOICE=1` report the four voice-only checks as intentionally disabled/healthy; configured voice retains the existing probes. Unreadable or malformed relevant configuration fails closed as `voice-config`.

Closes #2349.

## Review first

- `src/health-check.py`: `resolve_voice_health_config()` and `check_voice_stack()`
- `tests/health-check-voice-config.test.py`: configuration precedence and failure cases

## What behavior does this prove?

A supported text-only installation no longer stays permanently unhealthy merely because no Gemini voice credential is configured. A configured-but-broken voice agent is still a hard failure.

## Before / after evidence

Parent `fb0f8438c8c36aa74624269a8cdb3e7957c5c022`:

```text
$ python3 src/health-check.py 2>&1 | rg 'voice-agent|voice-watchers|voice-transport|bodhi-dist|issue\(s\) found'
✗ voice-agent                    down         port 9900
⚠ voice-watchers                 warn         voice-agent down
⚠ voice-transport                warn         voice-agent down
⚠ bodhi-dist                     warn         no bodhi artifact found (voice disabled but still probed)
4 issue(s) found:
  - voice-agent: down (port 9900)
```

HEAD `5cce4637` on the same credential-free host:

```text
$ python3 src/health-check.py 2>&1 | rg 'voice-agent|voice-watchers|voice-transport|bodhi-dist|issue\(s\) found'
✓ voice-agent                    ok           disabled (no Gemini voice credential configured)
✓ voice-watchers                 ok           disabled (no Gemini voice credential configured)
✓ voice-transport                ok           disabled (no Gemini voice credential configured)
✓ bodhi-dist                     ok           disabled (no Gemini voice credential configured)
3 issue(s) found:
```

The other three pre-existing health failures in the isolated worktree are unrelated to voice configuration.

## Tests

```text
$ python3 tests/health-check-voice-config.test.py
.......
Ran 7 tests in 0.171s
OK

$ python3 tests/health-check-bridge-log-content.test.py
PASS — bridge_log_content_status tests

$ python3 tests/health-check-bridge-skip.test.py
PASS — _should_skip_bridge tests

$ python3 tests/health-check-comm-sweep-freshness.test.py
Ran 6 tests in 0.224s
OK

$ python3 tests/health-check-fix-down-bridges.test.py
All fix_down_bridges tests passed.

$ python3 -m py_compile src/health-check.py tests/health-check-voice-config.test.py
$ git diff --check origin/main...HEAD
$ python3 scripts/review-checks.py
```

All commands exited 0.

## Edge cases

- no `.env` and empty `.env`;
- either Gemini voice credential;
- explicit `SKIP_VOICE=1`;
- process environment overriding file values, including explicit empty keys;
- unreadable config, malformed quotes/assignments, and invalid `SKIP_VOICE`;
- configured voice still executes stale-code, watcher, transport, and Bodhi probes.

## Live-path evidence

N/A. This changes health reporting only; it does not alter startup, service lifecycle, bridge delivery, network behavior, or voice sessions. The real host health command above exercises the production reporting path.

## Migrations, config, permissions, rollback, or deployment risks

No migration, new permission, or config key. The main risk is incorrectly hiding a configured voice outage; environment precedence, malformed-config failure, and the unchanged configured probe path are covered explicitly. Rollback is a single commit revert.

## Hardcoded-path check

`git diff origin/main...HEAD` adds no host-specific paths or inline path fallbacks. The implementation uses the existing canonical `_resolve_dotenv()` helper.

## Known gaps / follow-up

N/A.

### CODE REVIEW

**Review by marklysze — APPROVED** [review]
Reviewed the diff and checked it out to verify (detached, then reset — no branch left behind).

- `resolve_voice_health_config()` reads `SKIP_VOICE`/`GEMINI_VOICE_API_KEY`/`GEMINI_API_KEY` with correct process-env > dotenv precedence, fails closed (`enabled: True` + `error`) on unreadable file, malformed bare-key assignment, unparseable value, or invalid `SKIP_VOICE`. `check_voice_stack()` correctly short-circuits to three "ok/disabled" checks only when genuinely unconfigured, and preserves the full port/watcher/transport probe path when a credential is present — confirmed by reading `run_all_checks()`'s replacement of the old inline block with `check_voice_stack()`.
- Ran `python3 tests/health-check-voice-config.test.py` at HEAD (ba050c0): 6/6 pass.
- Cross-checked against my own host's `.env` (has `GEMINI_API_KEY` set): under this logic voice health stays required there, matching the existing `ok`/port-9900 result I saw in today's `health-check.py` run — no regression for the configured case.
- No hardcoded paths added; uses the existing `_resolve_dotenv()` helper.

LGTM.

**Review by qingyun-wu — CHANGES_REQUESTED** [review]
Blocking on current head `ba050c08ac7fff0343a038e72ba091a4cceb2d3c`.

`run_all_checks()` now routes the three voice process probes through `check_voice_stack()`, but it still unconditionally appends `check_bodhi_dist()` immediately afterward. That keeps disabled/text-only voice installs noisy: with no `SKIP_VOICE` issue and no Gemini voice key, the voice-agent/watchers/transport entries become `ok`, but a host without the voice dependency or bundled `dist/voice-agent.js` still gets a `bodhi-dist` warning telling it to install/build voice artifacts. `bodhi-dist` only validates the voice agent's Gemini realtime transport, so it should be gated by the same resolved voice-enabled state, or folded into `check_voice_stack()` with a disabled result. Please add coverage that exercises the actual `run_all_checks()`/bodhi path under disabled voice config, not only the three process probes.

Focused verification: `git diff --check`, `py_compile` for the changed Python files, and `python3 tests/health-check-voice-config.test.py` passed locally. GitHub checks are green, including ruff and the clean-install test job.

Reviewed by Qingyun's Personal Codex.

**Comment by john-the-dev** [comment]
Fixed the disabled Bodhi-probe blocker in `5cce4637`. `check_voice_stack()` now returns `bodhi-dist: ok / disabled` without calling `check_bodhi_dist()` when voice is disabled, while configured/error paths still run the real artifact probe. The new `run_all_checks()` regression mocks the Bodhi probe and asserts it is not called; focused coverage is now 7/7, nearby health suites still pass, and the real credential-free output shows all four voice-only checks disabled/ok. Please re-review this exact head and replace the stale REQUEST_CHANGES state once refreshed CI is green.

File: src/health-check.py [hunk 1]
```diff
@@ -104,6 +104,117 @@ def _resolve_dotenv() -> Path:
...
+            file_values[key] = parsed[0] if parsed else ""
+
+    def effective(key: str) -> str:
+        value = env[key] if key in env else file_values.get(key, "")
+        return str(value).strip()
+
+    skip_voice = effective("SKIP_VOICE")
+    if skip_voice not in ("", "0", "1"):
+        return {"enabled": True, "error": f"invalid SKIP_VOICE={skip_voice!r}"}
+    if skip_voice == "1":
```
Comment (qingyun-wu) [inline src/health-check.py:171]: [P1 blocking] This precedence does not match `configure_startup_runtime()`: when a Gemini key exists, startup unsets `SKIP_VOICE` and launches voice, but health-check returns disabled here. With both values in `.env`, the exact-head repro reports `startup_SKIPPED=0` and `health_enabled=False`, so a broken configured voice agent can be hidden as healthy. Please share/match the startup rule and add a cross-path regression.

**Review by qingyun-wu — CHANGES_REQUESTED** [review]
Changes requested on exact head `5cce463786f164e680309d9170f8b2aa88cc71bd`.

Blocking finding: health reporting still disagrees with the production startup decision when `SKIP_VOICE=1` and a Gemini key are both present. `configure_startup_runtime()` in `src/startup-runtime.sh:11-16` unsets `SKIP_VOICE` whenever either key is configured, so startup launches and verifies the voice stack. `resolve_voice_health_config()` instead returns disabled at `src/health-check.py:169-170`, which makes all four voice-only checks green without probing them. The worst case is a configured, broken voice agent being hidden as intentionally disabled.

Exact-head repro with `.env` containing `SKIP_VOICE=1` and `GEMINI_API_KEY=configured-key`:

```text
startup_SKIPPED=0
health_enabled= False
```

Please make health-check and startup share or match one precedence rule, and cover the mixed key + skip case across both paths.

Focused validation otherwise passes: 7 voice-config tests, `tests/startup-voice-optional.test.sh`, four nearby health suites, Python compilation with an isolated bytecode cache, `git diff --check`, repository review checks, and all current GitHub CI/CLA checks.

Reviewed by Qingyun's Personal Codex.

**Review by qingyun-wu — CHANGES_REQUESTED** [review]
Changes requested on exact head `fc8eec606258e4b2585459626995e289bb13bf10` after the main merge. The merge adds #2348 only; the PR-owned voice configuration code and the blocker are unchanged.

Current-head repro with `.env` containing `SKIP_VOICE=1` and `GEMINI_API_KEY=configured-key` still produces:

```text
startup_SKIPPED=0
health_enabled= False
```

`src/startup-runtime.sh:11-16` therefore launches voice, while `src/health-check.py:169-170` suppresses all voice probes as disabled. This can hide a configured voice outage. Please align/share the precedence and cover the mixed key + skip case across startup and health-check.

Current-head validation passes: 7 voice-config tests, startup optional-voice test, 25 stuck-loop/retention checks, Python compilation, diff/path checks. The behavioral mismatch remains the only blocker.

Reviewed by Qingyun's Personal Codex.

**Comment by john-the-dev** [comment]
Fixed the startup/health precedence blocker on exact head `1790c27b`. Health now checks either Gemini credential before interpreting `SKIP_VOICE`, matching `configure_startup_runtime()` which unsets the skip flag whenever a key is configured.

Added mixed `SKIP_VOICE=1` + key coverage in both `tests/health-check-voice-config.test.py` and the real `tests/startup-voice-optional.test.sh` path. Exact-head output:

```text
health_enabled=True
startup_SKIPPED=0
```

Validation: 7/7 voice-config, startup optional-voice PASS, 7/7 Bodhi artifact, 25/25 queue/stuck-loop, 5/5 summary tests, Python compile, `git diff --check`, and repository review checks/hardcoded-path scan PASS. Please re-review exact head `1790c27b` and replace the stale REQUEST_CHANGES state once refreshed CI is green.

**Review by qingyun-wu — APPROVED** [review]
Approved current head `1790c27b04f90728c6cf4c1309287833e949699a`.

No blocking findings. The latest update aligns health-check with startup precedence for the mixed `SKIP_VOICE=1` plus configured Gemini key case, keeps voice/Bodhi probes required whenever a voice-capable credential is configured, and still short-circuits voice-only checks to ok/disabled for genuine text-only installs.

Focused verification passed: Python compilation for the changed health-check/test files, `tests/health-check-voice-config.test.py` (7/7), `tests/startup-voice-optional.test.sh`, `tests/health-check-bodhi-dist.test.py` (7/7), `tests/health-check-stuck-loop.test.py` (25 cases), `tests/health-check-summary-warnings.test.py` (5/5), and all refreshed GitHub checks are green.

Reviewed by Qingyun's Personal Codex.

**Review by sonichi — APPROVED** [review]
Approving — config-aware voice checks; fail-safe on invalid SKIP_VOICE; matches startup precedence with a parity test.
